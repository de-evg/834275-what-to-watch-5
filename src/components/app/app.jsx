import React from "react";
import PropTypes from "prop-types";
import {BrowserRouter, Switch, Route} from "react-router-dom";

import MainScreen from "../main-screen/main-screen";
import AuthScreen from "../auth-screen/auth-screen";
import MyListScreen from "../my-list-screen/my-list-screen";
import MovieScreen from "../movie-screen/movie-screen";
import AddReviewScreen from "../add-review-screen/add-review-screen";
import PlayerScreen from "../player-screen/player-screen";


const App = (props) => {
  const {promoTitle, promoGenre, promoRelease, films} = props;
  return (
    <BrowserRouter>
      <Switch>
        <Route exact path="/">
          <MainScreen
            promoTitle={promoTitle}
            promoGenre={promoGenre}
            promoRelease={promoRelease}
            films={films}
          />
        </Route>

        <Route exact path="/login">
          <AuthScreen />
        </Route>

        <Route exact path="/myList">
          <MyListScreen />
        </Route>

        <Route exact path="/films/:id">
          <MovieScreen />
        </Route>

        <Route exact path="/films/:id/review">
          <AddReviewScreen />
        </Route>

        <Route exact path="/player/:id">
          <PlayerScreen />
        </Route>
      </Switch>
    </BrowserRouter>
  );
};

App.propTypes = {
  promoTitle: PropTypes.string.isRequired,
  promoGenre: PropTypes.string.isRequired,
  promoRelease: PropTypes.number.isRequired
};

export default App;
