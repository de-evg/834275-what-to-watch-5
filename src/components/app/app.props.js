import PropTypes from "prop-types";

const appProps = {
  authorizationStatus: PropTypes.string.isRequired
};

export default appProps;
