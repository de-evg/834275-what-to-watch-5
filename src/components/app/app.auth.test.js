import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {App} from "./app";

describe(`Render App`, () => {
  const mockStore = configureStore([]);
  let store = null;
  let AppComponent = null;

  beforeEach(() => {
    store = mockStore({
      DATA: {
        currentGenre: `All genres`,
        movies: [],
        filteredMovies: [],
        genres: [],
        promo: {
          title: ``,
          genre: ``,
          release: 0,
          posterURL: ``,
          previewURL: ``,
          id: 0,
          isInWatchList: false
        },
        movie: ``,
        movieIsLoaded: false
      },
      REVIEW: {
        reviews: []
      },
      USER: {
        authorizationStatus: `AUTH`,
        userID: ``,
        userName: ``,
        userAvatar: ``,
        userEmail: ``,
        authError: false
      }
    });

    store.dispatch = jest.fn();

    AppComponent = renderer
        .create(
            <Provider store={store}>
              <App
                authorizationStatus={``}
              />)
            </Provider>
        );
  });

  it(`Should App connected to store render correctly`, () => {
    expect(AppComponent.toJSON()).toMatchSnapshot();
  });
});

