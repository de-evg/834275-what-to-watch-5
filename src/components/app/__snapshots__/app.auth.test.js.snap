// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App Should App connected to store render correctly 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt=""
        src=""
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <a
            href="/myList"
            onClick={[Function]}
          >
            <img
              alt="User avatar"
              height="63"
              src=""
              width="63"
            />
          </a>
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt=""
            height="327"
            src=""
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              
            </span>
            <span
              className="movie-card__year"
            >
              0
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/0"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      />
      <div
        className="catalog__movies-list"
      />
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
  ")",
]
`;
